Casks/c/crashplan.rb

```diff
@@ -0,0 +1,34 @@
+cask "crashplan" do
+  version "11.3.1,3"
+  sha256 "d4c1f96a7d207536f6a6877da717f282778610b7961fb41c60ccb467120ce5fc"
+
+  url "https://download.crashplan.com/installs/agent/cloud/#{version.csv.first}/#{version.csv.second}/install/CrashPlan_#{version.csv.first}_#{version.csv.second}_Mac.dmg"
+  name "CrashPlan"
+  desc "Backup and recovery software"
+  homepage "https://www.crashplan.com/"
+
+  livecheck do
+    url "https://download.crashplan.com/installs/agent/latest-mac.dmg"
+    strategy :header_match do |headers|
+      match = headers["location"].match(%r{/(\d+(?:\.\d+)+)/(\d+)/})
+      "#{match[1]},#{match[2]}" if match
```

Comment:
```suggestion
    regex(/CrashPlan[._-]v?(\d+(?:[._]\d+)+)/i)
    strategy :header_match do |headers, regex|
      match = headers["location"]&.match(regex)
      next if match.blank?

      match[1].tr("_", ",")
```

A few tweaks:

* This should use `#regex` and pass the regex into the strategy block.
* Matching the version and secondary number from the path (`.../11.3.1/3/...`) works but matching the version from the filename is a more typical pattern and we can easily set this up so that it will work even if the secondary number is omitted in the future. That said, bumping the cask would still require changes to the `url` format unless we conditionally include `version.csv.second` (like in the `jasp` cask).
* The `location` header isn't guaranteed to exist, so I tend to use the safe navigation operator (`&.`) after values like this.
* The `if match` setup is tidier but the early return pattern ends up being nicer in some scenarios (e.g., a ternary return value), so we've kind of standardized on this `next if match.blank?` approach.

Reply:
Thank you @samford !
That, of course, is a much nicer approach. I will remember this for the future!